crates/hyfetch/src/bin/hyfetch.rs

```diff
@@ -0,0 +1,1034 @@
+        debug!(?color_mode, "selected color mode");
+        update_title(&mut title, &mut option_counter, ttl, color_mode.as_ref());
+        color_mode
+    };
+
+    //////////////////////////////
+    // 2. Select theme (light/dark mode)
+
+    let select_theme = || -> Result<(TerminalTheme, &str)> {
+        if let Some(det_bg) = det_bg {
+            return Ok((det_bg.theme(), "Detected background color"));
+        }
+
+        clear_screen(Some(&title), color_mode, debug_mode).context("failed to clear screen")?;
+
+        print_title_prompt(
+            option_counter,
+            "Is your terminal in &blight mode&~ or &4dark mode&~?",
+            color_mode,
+        )
+        .context("failed to print title prompt")?;
+        let choice = literal_input(
+            "",
+            TerminalTheme::VARIANTS,
+            TerminalTheme::Dark.as_ref(),
+            true,
+            color_mode,
+        )
+        .context("failed to ask for choice input")?;
+        Ok((
+            choice.parse().expect("selected theme should be valid"),
+            "Selected background color",
+        ))
+    };
+
+    let theme = {
+        let (theme, ttl) = select_theme().context("failed to select theme")?;
+        debug!(?theme, "selected theme");
+        update_title(&mut title, &mut option_counter, ttl, theme.as_ref());
+        theme
+    };
+
+    //////////////////////////////
+    // 3. Choose preset
+
+    // Create flag lines
+    let mut flags = Vec::with_capacity(Preset::COUNT);
+    let spacing = {
+        let spacing = <Preset as VariantNames>::VARIANTS
+            .iter()
+            .map(|name| name.chars().count())
+            .max()
+            .expect("preset name iterator should not be empty");
+        let spacing = u8::try_from(spacing).expect("`spacing` should fit in `u8`");
+        cmp::max(spacing, 20)
+    };
+    for preset in <Preset as VariantArray>::VARIANTS {
+        let color_profile = preset.color_profile();
+        let flag = color_profile
+            .color_text(
+                " ".repeat(usize::from(spacing)),
+                color_mode,
+                ForegroundBackground::Background,
+                false,
+            )
+            .with_context(|| format!("failed to color flag using preset: {preset:?}"))?;
+        let name = format!(
+            "{name:^spacing$}",
+            name = preset.as_ref(),
+            spacing = usize::from(spacing)
+        );
+        flags.push([name, flag.clone(), flag.clone(), flag]);
+    }
+
+    // Calculate flags per row
+    let (flags_per_row, rows_per_page) = {
+        let (Width(term_w), Height(term_h)) =
+            terminal_size().context("failed to get terminal size")?;
+        let flags_per_row = term_w / (u16::from(spacing) + 2);
+        let flags_per_row =
+            u8::try_from(flags_per_row).expect("`flags_per_row` should fit in `u8`");
+        let rows_per_page = cmp::max(1, (term_h - 13) / 5);
```

Comment:
Also here, when the terminal is only a few lines in height, this will underflow (unsigned), and then later panic on the u8 conversion.